File: tests/fixtures/swift/basic.swift
// ============================================================
// Swift Test: Basic Symbols
// ============================================================

import Foundation

// ------------------------------------------------------------
// Protocol
// ------------------------------------------------------------
protocol Identifiable {
    var id: String { get }
    func identify() -> String
}

protocol Cacheable {
    func cache()
    func invalidate()
}

// ------------------------------------------------------------
// Class
// ------------------------------------------------------------
class User: Identifiable {
    let id: String
    var name: String
    var email: String

    init(id: String, name: String, email: String) {
        self.id = id
        self.name = name
        self.email = email
    }

    func identify() -> String {
        return id
    }

    func validate() -> Bool {
        return !name.isEmpty && email.contains("@")
    }
}

// ------------------------------------------------------------
// Struct
// ------------------------------------------------------------
struct Point {
    var x: Double
    var y: Double

    func distance(to other: Point) -> Double {
        let dx = x - other.x
        let dy = y - other.y
        return (dx * dx + dy * dy).squareRoot()
    }
}

struct Address {
    let street: String
    let city: String
    let country: String
}

// ------------------------------------------------------------
// Enum
// ------------------------------------------------------------
enum Status {
    case active
    case inactive
    case pending
}

enum Result<T> {
    case success(T)
    case failure(Error)
}

// ------------------------------------------------------------
// Extension
// ------------------------------------------------------------
extension User {
    var displayName: String {
        return "\(name) <\(email)>"
    }
}

// ------------------------------------------------------------
// Typealias
// ------------------------------------------------------------
typealias UserID = String
typealias UserHandler = (User) -> Void

// ------------------------------------------------------------
// Functions
// ------------------------------------------------------------
func createUser(name: String, email: String) -> User {
    return User(id: UUID().uuidString, name: name, email: email)
}

func greet(_ user: User) {
    print("Hello, \(user.name)!")
}

private func internalHelper() -> Bool {
    return true
}

File: tests/fixtures/kotlin/basic.kt
// ============================================================
// Kotlin Test: Basic Symbols
// ============================================================

package com.example.test

// ------------------------------------------------------------
// Interface
// ------------------------------------------------------------
interface Identifiable {
    val id: String
    fun identify(): String
}

interface Cacheable {
    fun cache()
    fun invalidate()
}

// ------------------------------------------------------------
// Class
// ------------------------------------------------------------
class User(
    override val id: String,
    val name: String,
    val email: String
) : Identifiable {

    override fun identify(): String = id

    fun validate(): Boolean {
        return name.isNotEmpty() && email.contains("@")
    }

    companion object {
        fun create(name: String, email: String): User {
            return User(java.util.UUID.randomUUID().toString(), name, email)
        }
    }
}

// ------------------------------------------------------------
// Data Class
// ------------------------------------------------------------
data class Address(
    val street: String,
    val city: String,
    val country: String
)

data class Point(val x: Double, val y: Double) {
    fun distance(other: Point): Double {
        val dx = x - other.x
        val dy = y - other.y
        return kotlin.math.sqrt(dx * dx + dy * dy)
    }
}

// ------------------------------------------------------------
// Enum
// ------------------------------------------------------------
enum class Status {
    ACTIVE,
    INACTIVE,
    PENDING
}

enum class UserRole(val level: Int) {
    ADMIN(3),
    USER(2),
    GUEST(1)
}

// ------------------------------------------------------------
// Object (Singleton)
// ------------------------------------------------------------
object UserRepository {
    private val users = mutableListOf<User>()

    fun add(user: User) {
        users.add(user)
    }

    fun find(id: String): User? {
        return users.find { it.id == id }
    }
}

// ------------------------------------------------------------
// Sealed Class
// ------------------------------------------------------------
sealed class Result<out T> {
    data class Success<T>(val data: T) : Result<T>()
    data class Error(val message: String) : Result<Nothing>()
}

// ------------------------------------------------------------
// Extension Function
// ------------------------------------------------------------
fun User.displayName(): String {
    return "$name <$email>"
}

// ------------------------------------------------------------
// Top-level Functions
// ------------------------------------------------------------
fun createUser(name: String, email: String): User {
    return User.create(name, email)
}

fun greet(user: User) {
    println("Hello, ${user.name}!")
}

private fun internalHelper(): Boolean = true

// ------------------------------------------------------------
// Top-level Properties
// ------------------------------------------------------------
val DEFAULT_NAME = "Guest"
const val MAX_USERS = 100
